libsolidity/codegen/YulUtilFunctions.cpp

```diff
@@ -630,6 +630,104 @@ string YulUtilFunctions::overflowCheckedIntExpFunction(
 	});
 }
 
+string YulUtilFunctions::overflowCheckedIntLiteralExpFunction(
+	RationalNumberType const& _baseType,
+	IntegerType const& _exponentType,
+	IntegerType const& _commonType
+)
+{
+	solAssert(!_exponentType.isSigned(), "");
+	solAssert(_baseType.isNegative() == _commonType.isSigned(), "");
+	solAssert(_commonType.numBits() == 256, "");
+
+	// Calculates the upperbound for exponentiation, that is, calculate `b`, such that
+	// _base**b <= _maxValue and _base**(b + 1) > _maxValue
+	auto findExponentUpperbound = [](bigint const _base, bigint const _maxValue) -> unsigned
```

Comment:
Technically this is same as `floor(log_2(_base)/log_2(_maxValue))` in 'exact' arithmetic, but I didn't want to deal with floating point errors (during the `log_2` computation and division) causing issues with the calculation.

Reply:
What if `_base` is zero or one?

Reply:
These cases are buggy. I'll fix them. Thanks!